AEPMessaging/Sources/UI/ContentCards/UIElement/AEPImage/AEPAsyncImageView.swift

```diff
@@ -97,4 +92,16 @@ struct AEPAsyncImage<Content>: View where Content: View {
             }
         }.resume()
     }
+
+    /// Determines the appropriate URL for the image based on the device's color scheme.
+    /// - Parameter colorScheme: The `ColorScheme` that determines whether to use light or dark mode URL.
+    ///
+    /// - Returns: The URL to be used for the image.
+    private func themeBasedURL(_ colorScheme: ColorScheme) -> URL {
```

Comment:
since this is private, can we just use the private var `colorScheme` directly, rather than having an input parameter?

Reply:
Getting from the private variable wont work here. Because when we re-render AEPAysncImageView  [onChange](https://github.com/adobe/aepsdk-messaging-ios/pull/328/files#diff-069a68fd6c3c819eae1061242b235b471179d3328aee4ce070c50c1182e0096fR59)

We will need the newColorScheme to be injected to get the appropriate url

the private var colorscheme still isn't updated by the time onChange is called.

Reply:
We see the following behaviour

https://github.com/user-attachments/assets/f01d672f-3817-424f-91dd-a075da1692a8